src/AppInstallerCommonCore/NameNormalization.cpp

```diff
@@ -0,0 +1,441 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "Public/winget/NameNormalization.h"
+#include "Public/AppInstallerStrings.h"
+#include "Public/winget/Regex.h"
+
+
+namespace AppInstaller::Utility
+{
+    namespace
+    {
+        struct InterimNameNormalizationResult
+        {
+            std::wstring Name;
+            Architecture Architecture;
+            std::wstring Locale;
+        };
+
+        struct InterimPublisherNormalizationResult
+        {
+            std::wstring Publisher;
+        };
+
+        // To maintain consistency, changes that result in different output must be done in a new version.
+        // This can potentially be ignored (if thought through) when the changes will only increase the
+        // number of matches being made, with no impact to existing matches. For instance, removing an
+        // arbitrary new processor architecture from names would hopefully only affect existing packages
+        // that were not matching properly. Fixing a bug that was causing bad strings to be produced would
+        // be impactful, and thus should likely result in a new iteration.
+        class NormalizationInitial : public details::INameNormalizer
+        {
+            static std::wstring PrepareForValidation(std::string_view value)
+            {
+                std::wstring result = Utility::Normalize(ConvertToUTF16(value));
+                Trim(result);
+                size_t atPos = result.find(L"@@", 3);

```

Comment:
>@@ [](start = 45, length = 2)

what's special with this?

Reply:
I have no idea; the starting point for this code was something doing similar work, and I left the majority of it alone.